oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/worker/MetricsPersistentWorker.java

```diff
@@ -174,21 +175,25 @@ private void flushDataToStorage(List<Metrics> metricsList,
                     /*
                      * Merge metrics into cachedMetrics, change only happens inside cachedMetrics.
                      */
-                    cachedMetrics.combine(metrics);
-                    cachedMetrics.calculate();
-                    prepareRequests.add(metricsDAO.prepareBatchUpdate(model, cachedMetrics));
-                    nextWorker(cachedMetrics);
+                    isAbandoned = !cachedMetrics.combine(metrics);
+                    if (isAbandoned) {
```

Comment:
```suggestion
                    if (!isAbandoned) {
```
Is this reverse?

Reply:
Or should we see `if isBandoned; continue`?

Reply:
This should make the metrics never updated again, from my reading.

Reply:
> Or should we see `if isBandoned; continue`?

🤦🏻‍♂️ my stupid mistake, fixed